homeassistant/components/recorder/migration.py

```diff
@@ -518,6 +518,29 @@ def _apply_update(engine, session, new_version, old_version):  # noqa: C901
                             "CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci"
                         )
                     )
+    elif new_version == 22:
+        # Recreate the statistics, statistics meta and statistics meta tables with Identity columns
+        #
+        # Order matters! Statistics has a relation with StatisticsMeta,
+        # so statistics need to be deleted before meta (or in pair depending
+        # on the SQL backend); and meta needs to be created before statistics.
+        if (
```

Comment:
We only want to do this if the engine dialect is oracle.

Reply:
What about other engines which support Identity colums e.g. Postrges, MSSQL? The state of database will not reflect what is in models.py, or did I miss something?

Reply:
You mention in the PR description that:
> Lack of Identity column breaks recorder functionality on Oracle DB

There have been no complaints saying that the recorder functionality is broken for any other database engines.

We should only drop tables if they are currently not working because the `Identity()` was missing and there's no way to update the tables to fix that.

Reply:
Can you clarify what the difference between these options for defining a column are:
`Column(Integer, Identity(), primary_key=True)`
`Column(Integer, primary_key=True)`

The SQLAlchemy documentation is not very clear, but the way I understand it, `Identity()` is optional and can be used to override the default behavior of a primary key?

Reply:
I'm sorry I was wrong.

I did some tests and for DB engines other than Oracle there is not much change how tables are created with or without Identity column e.g. for MySQL AUTO_INCREMENT will be added with or without Identity parameter, for PostgreSQL 'Default' field will be different i.e. without Identity() it will be "nextval('table_id_seq'::regclass)", with Identity() it will be "generated by default as identity", but under the hood the same sequence will be created "nextval('table_id_seq'::regclass)" and so on, so in fact it affects Oracle DB only.

Sorry for any inconveniences, I will update migration logic to be applied for engine dialect oracle only.

Reply:
No problem at all, thanks a lot for double checking 👍